src/Discord.Net.SlashCommands/TypeConverters/EnumConverter.cs

```diff
@@ -0,0 +1,42 @@
+using Discord.WebSocket;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Discord.SlashCommands
+{
+    internal sealed class EnumConverter<T> : TypeConverter<T> where T : Enum
+    {
+        public override ApplicationCommandOptionType GetDiscordType ( ) => ApplicationCommandOptionType.String;
+        public override Task<TypeConverterResult> ReadAsync (ISlashCommandContext context, SocketSlashCommandDataOption option, IServiceProvider services)
```

Comment:
Value task this

Reply:
User defined TypeConverters should be able to pass additional info to the InteractionService in case of failure